### PR TITLE
refactor: tailwind for icon component

### PR DESCRIPTION
**Pull request by patricio0312rev**
# [[extension] icon component tailwind refactor](https://app.clickup.com/t/86drqmg9n)

## Summary

- Depends on #44  and #38 
- Icon component has been refactored and now works with tailwind classes.
- All the implementations of this component have been refactored.

## Checklist

- [ ] I checked that both `pnpm dev` and `pnpm dev:bare` work as intended
- [ ] I checked the basic extension interactions and made sure wallet selection works
- [ ] I checked my UI changes against the design and there are no notable differences, including responsiveness
- [ ] I checked my (code) changes for obvious issues, debug statements and commented code
- [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
- [ ] I added a short description on how to test this PR _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

**Comment by ItsANameToo** [comment]
Draft until #44 is merged

File: src/shared/components/icon/index.tsx [hunk 1]
```diff
@@ -1,32 +1,23 @@
...
-import styled from 'styled-components';
-import { color, ColorProps, layout, LayoutProps } from 'styled-system';
 import { IconDefinition, IconSvg } from './index.generated';
-import { Theme } from '@/shared/theme';
+import cn from 'classnames';
 export * from './index.generated';
 
 type IconProps = {
     icon: IconDefinition;
     style?: React.CSSProperties;
```
Comment (shahin-hq) [inline src/shared/components/icon/index.tsx:8]: I think we can remove `style` as well since we are using classes to style things

**Review by alfonsobries — CHANGES_REQUESTED** [review]
Not sure if introduced on this pr but the back button is invisible apparently everywhere, on the screenshot you can see the "hover" style, if I dont hover it is invisible

image

**Comment by alfonsobries** [comment]
connect to app icon is now lighter

Before

image

After
image

**Comment by alfonsobries** [comment]
rest of actions icons are lighter as well
image

**Comment by alfonsobries** [comment]
main logo is lighter 
image

**Comment by alfonsobries** [comment]
another that is lighter
image

File: src/shared/components/icon/index.tsx [hunk 2]
```diff
@@ -1,32 +1,16 @@
...
-export const Icon: FC<IconProps> = ({ icon, style, className, ...rest }) => {
+export const Icon = ({ icon, className, ...rest }: IconProps) => {
     return (
-        <StyledIcon
-            {...rest}
-            {...{ style }}
-            className={'Icon ' + (className ? className : '')}
-            role='img'
-        >
+        <span {...rest} className={cn('block', 'Icon ' + (className ? className : ''))} role='img'>
```
[the remark this answers is not in the document]
Comment (ItsANameToo) [reply src/shared/components/icon/index.tsx]: removed it

**Pull request merged**